codedigger/lists/views.py

```diff
@@ -722,6 +722,18 @@ def get(self, request, slug):
                                 problem_qs, isCompleted)
         return response.Response(res)
 
+class SearchUserlistView(generics.ListAPIView):
+    permission_classes = [AuthenticatedOrReadOnly]
+    serializer_class = [GetUserlistSerializer]
+
+    def get(self, request):
+        param = request.GET.get('q').lower()
+        if List.objects.filter(Q(name__icontains = param) & Q(public=True)):
```

Comment:
Redundant two times Filter. If the list name doesn't exist, that is not an error. Return empty list that is by default.